edgelet/iotedged/src/lib.rs

```diff
@@ -689,7 +689,11 @@ fn vol_mount_uri(config: &mut DockerConfig, uris: &[&Url]) -> Result<(), Error>
             let path = path
                 .parent()
                 .ok_or_else(|| ErrorKind::InvalidUri(uri.to_string()))?;
-            let path = path.to_str().unwrap().to_string();
+            let path = path
+                .to_str()
+                .expect(
+                    "URL path should be percent-encoded ASCII, but was found to be invalid UTF-8",
```

Comment:
Well, yes it's percent-encoded ASCII, but the error is that the *path* that was encoded using percent-encoding turned out to not be valid UTF-8 after it was decoded.

Perhaps something like `"URL points to a path that cannot be represented in UTF-8."`